sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/StringUtilsSuite.scala

```diff
@@ -43,4 +43,70 @@ class StringUtilsSuite extends SparkFunSuite {
     assert(filterPattern(names, " a. ") === Seq("a1", "a2"))
     assert(filterPattern(names, " d* ") === Nil)
   }
+
+  test("split a SQL") {
```

Comment:
The test case coverage is still not enough. For example, adding some test cases from https://github.com/apache/hive/commit/65a65826a0d351a3d918bdb98595bdd106d37adb#diff-6182a9d1d63c707dff0ecd4e6a025fd2 ?